### PR TITLE
finished 121

### PR DESCRIPTION
**Pull request by JoannaCode**
@FreeTymeKiyan

### CODE REVIEW

File: Easy/BestTimetoBuyandSellStock1.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * If you were only permitted to complete at most one transaction (ie, buy one and sell one share of the stock), design an algorithm to find the maximum profit.
+ * Tags:  Array Dynamic Programming
+ * Similar Problems:  (M) Maximum Subarray (M) Best Time to Buy and Sell Stock II (H) Best Time to Buy and Sell Stock III (H) Best Time to Buy and Sell Stock IV (M) Best Time to Buy and Sell Stock with Cooldown
+ * O(n) time, O(1) space
+ * @author chenshuna
+ */ 
+
+public class BestTimetoBuyandSellStock {
+    public static int maxProfit(int[] prices) {
+        if(prices == null || prices.length == 0 || prices.length == 1){
```
Comment (FreeTymeKiyan) [inline Easy/BestTimetoBuyandSellStock1.java]: `prices.length == 0 || prices.length == 1` <==> `prices.length < 2`

File: Easy/BestTimetoBuyandSellStock1.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            return 0;
+        }
+        int max = 0;
+        int minPrice = prices[0];
+        for(int i = 0; i < prices.length; i++){
```
Comment (FreeTymeKiyan) [inline Easy/BestTimetoBuyandSellStock1.java]: i can start from 1 as you make sure that `prices.length > 1` and max won't be updated until 1

File: Easy/BestTimetoBuyandSellStock1.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            minPrice = Math.min(minPrice, prices[i]);
+            max = Math.max(max,  prices[i] - minPrice);
```
Comment (FreeTymeKiyan) [inline Easy/BestTimetoBuyandSellStock1.java]: Just share something fun here. Here `max` is the difference of the maximum and the minimum of the previous prices. Think about this, if `prices[i -1] > prices[i]`, then `maximum of the previous prices > prices[i]`. Then `prices[i] - minPrice` won't be larger than `max`. 

That means, max will only be updated when `prices[i] > prices[i - 1]`.

**Comment by JoannaCode** [comment]
@FreeTymeKiyan  fixed!

**Comment by FreeTymeKiyan** [comment]
LGTM

**Pull request merged**